### ISSUE TITLE
Package type for container packages should be explicitly declared as well.

### ISSUE THREAD

**Issue by kwin** [AdobeDocs/experience-manager-cloud-service.en#19]
In help/implementing/developing/introduction/aem-project-content-package-structure.md it is incorrectly stated (https://docs.adobe.com/content/help/en/experience-manager-cloud-service/implementing/developing/aem-project-content-package-structure.html#package-types)
> Container packages must not have a packageType set.

This is not correct, even container package types should be explicity given as otherwise validation related to package types cannot be performed correctly(https://jackrabbit.apache.org/filevault/validation.html)

**Comment by davidjgonzalez** [comment]
@kwin 

The reason we recommend not setting the packageType today is we (AEM as a Cloud Service) want the flexibility in the future to relax some of the constraints imposed by the filevault plugin (which would otherwsie require users to remove that packageType in their project to make the changes work). Today, they are "technically compliant" filevault container packageType packages, but conceptually they are really "AEM container packages" that may evolve apart from filevault's constraints in the future; we dont want to set people's projects to break if/when that future arrives.

It's a little confusing and talks about possible futures, but that was the underlying thought.

TL;DR dont set the container packageType. If you do, you may have to remove it sometime in the future.

**Comment by kwin** [comment]
If you don't set an explicit packageType the filevault-package-maven-plugin tries to guess it! That does not work really reliable (https://issues.apache.org/jira/browse/JCRVLT-431). Therefore you will always end up with a specific packageType in the `META-INF/vault/properties.xml`. I strongly recommend setting it to `MIXED` in the case you don't want it to be a `CONTAINER`!

**Comment by davidjgonzalez** [comment]
@kwin ill log internal issues for this, as I'm not sure where Eng will want to address this - depending on what they want to do, we can update the docs, or something else!

**Comment by davidjgonzalez** [comment]
@kwin logged internally as GRANITE-29573

We will update the issue/docs/whatever we need once we hear back. Thanks for the heads up!

**Comment by kwin** [comment]
`MIXED` means to limitations so I guess this is what you want to set in this case!

**Comment by DominikSuess** [comment]
@kwin as far as I could quickly check the "all" package would match the latest clarifications we had on what container packages may be able. We anyhow currently struggle to bump the plugin version beyond 1.0.4 as the validation configuration seems not to work as expected. Due to legacy reasons, we don't want to enforce (but potentially warn) about moving install folders out to the container as we anyhow have the converter cutting those ones out of the package later on - yet somehow the reconfiguration of installer path regex to make it "blind" for those changes didn't work for me (where I would also be fine to reconfigure the severity of this "subaspect" of the packageType validator).  It would be great if you could provide examples (e.g. for wknd guide) how a low intrusive adoption of the latest version of the plugin might work.

**Comment by kwin** [comment]
"application" packages don't allow bundles! That is the current reality and comes directly from https://issues.apache.org/jira/browse/JCRVLT-170. If you want to use OSGi config/bundles in "application" packages you are violating the package type declaration. But as said in https://github.com/adobe/aem-project-archetype/issues/325#issuecomment-617719313 it may be possible to introduce a flag to the validator to relax that particullar check.

**Comment by bohnertchris** [comment]
CQDOC-16096
